api/logical.go

```diff
@@ -106,6 +90,41 @@ func (c *Logical) ReadWithDataWithContext(ctx context.Context, path string, data
 	return ParseSecret(resp.Body)
 }
 
+func (c *Logical) ReadRaw(path string) (*Response, error) {
+	return c.ReadRawWithData(path, nil)
+}
+
+func (c *Logical) ReadRawWithData(path string, data map[string][]string) (*Response, error) {
+	return c.ReadRawWithDataWithContext(context.Background(), path, data)
+}
+
+func (c *Logical) ReadRawWithDataWithContext(ctx context.Context, path string, data map[string][]string) (*Response, error) {
+	ctx, cancelFunc := c.c.withConfiguredTimeout(ctx)
+	defer cancelFunc()
+
+	return c.readRawWithDataWithContext(ctx, path, data)
+}
+
+func (c *Logical) readRawWithDataWithContext(ctx context.Context, path string, data map[string][]string) (*Response, error) {
+	r := c.c.NewRequest(http.MethodGet, "/v1/"+path)
+
+	var values url.Values
+	for k, v := range data {
+		if values == nil {
+			values = make(url.Values)
+		}
+		for _, val := range v {
+			values.Add(k, val)
+		}
+	}
+
+	if values != nil {
+		r.Params = values
+	}
+
+	return c.c.RawRequestWithContext(ctx, r)
```

Comment:
Is it possible we aren't closing the response body from this call like we are elsewhere? Or have I missed the defer to Close()?

Reply:
nvm I see it now in the caller below...

Reply:
I wonder if we should instead be returning a `[]byte` instead of the `*Response` so the end-user doesn't have to worry about the deferred Closing and is a bit more disconnected from the implementation details?

Reply:
Hmmm, maybe. You'd do that on line 105? That might be more palatable, yeah.

Reply:
Yes sorry, within the the caller function in line 105.

Reply:
Makes sense, yeah.

Reply:
I don't like that though. How do we handle failures?

Reply:
In our CLI example, we don't need it, but in a generalized API (which this is part of), someone probably wants the raw `Response` rather than just the `[]byte` data as it discards useful info.

Reply:
Right I keep forgetting Secret has IsError and stuff as well in it... Okay comment withdrawn.